pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -49,173 +51,209 @@ import (
 	"istio.io/istio/pkg/workloadapi/security"
 )
 
-func TestAmbientIndex(t *testing.T) {
+const (
+	testNS   = "ns1"
+	systemNS = "istio-system"
+	testNW   = "testnetwork"
+	testC    = "cluster0"
+)
+
+func TestAmbientIndex_NetworkAndClusterIDs(t *testing.T) {
 	test.SetForTest(t, &features.EnableAmbientControllers, true)
-	cfg := memory.NewSyncController(memory.MakeSkipValidation(collections.PilotGatewayAPI()))
-	controller, fx := NewFakeControllerWithOptions(t, FakeControllerOptions{
-		ConfigController: cfg,
-		MeshWatcher:      mesh.NewFixedWatcher(&meshconfig.MeshConfig{RootNamespace: "istio-system"}),
-		ClusterID:        "cluster0",
-	})
-	controller.network = "testnetwork"
-	pc := clienttest.Wrap(t, controller.podsClient)
-	sc := clienttest.Wrap(t, controller.services)
-	cfg.RegisterEventHandler(gvk.AuthorizationPolicy, controller.AuthorizationPolicyHandler)
-	cfg.RegisterEventHandler(gvk.WorkloadEntry, controller.WorkloadEntryHandler)
-	cfg.RegisterEventHandler(gvk.PeerAuthentication, controller.PeerAuthenticationHandler)
-	go cfg.Run(test.NewStop(t))
 
-	addPods := func(ip string, name, sa string, labels map[string]string, annotations map[string]string) {
-		t.Helper()
-		pod := generatePod(ip, name, "ns1", sa, "node1", labels, annotations)
+	cases := []struct {
+		name    string
+		cluster cluster.ID
+		network network.ID
+	}{
+		{
+			name:    "values unset",
+			cluster: "",
+			network: "",
+		},
+		{
+			name:    "values set",
+			cluster: testC,
+			network: testNW,
+		},
+	}
 
-		p := pc.Get(name, pod.Namespace)
-		if p == nil {
-			// Apiserver doesn't allow Create to modify the pod status; in real world its a 2 part process
-			pod.Status = corev1.PodStatus{}
-			newPod := pc.Create(pod)
-			setPodReady(newPod)
-			newPod.Status.PodIP = ip
-			newPod.Status.PodIPs = []corev1.PodIP{
-				{
-					IP: ip,
-				},
-			}
-			newPod.Status.Phase = corev1.PodRunning
-			pc.UpdateStatus(newPod)
-		} else {
-			pc.Update(pod)
-		}
+	for _, c := range cases {
+		c := c
```

Comment:
fwiw this is not needed when you don't use t.Parallel

Reply:
yep, just a habit I'd gotten into. Removed it.